packages/playwright-core/src/server/injected/recorder/recorder.ts

```diff
@@ -1004,7 +1004,12 @@ export class Recorder {
     removeEventListeners(this._listeners);
     this._listeners = [
       addEventListener(this.document, 'click', event => this._onClick(event as MouseEvent), true),
-      addEventListener(this.document, 'auxclick', event => this._onClick(event as MouseEvent), true),
+      addEventListener(this.document, 'auxclick', event => {
+        // Right clicks are handled by 'contextmenu' event.
+        if ((event as MouseEvent).button === 2)
```

Comment:
I think this is not the best place for this check. We should bail out in the place that actually records, instead of here. This event is fanned out to the overlay and all kinds of tools, and the logic you put here is specific to `RecordActionTool`.